core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -194,6 +197,17 @@ private String extractClassPathResourceToTempLocation(final String hostPath) {
         return tmpLocation.getAbsolutePath();
     }
 
+    private File createTempDirectory() {
+        try {
+            if (SystemUtils.IS_OS_MAC) {
+                return Files.createTempDirectory(Paths.get(OS_MAC_TMP_DIR), TESTCONTAINERS_TMP_DIR_PREFIX).toFile();
+            }
+            return Files.createTempDirectory(TESTCONTAINERS_TMP_DIR_PREFIX).toFile();
+        } catch  (IOException e) {
```

Comment:
Can you please explain the conditions and circumstances, under which you expect this `Exception` to happen and how this return value solves the problem?

Reply:
For example user runs java with params `-Djava.io.tmpdir=<folder>` on linux/windows.

In cases, the folder does not exist we will get IOException.

I think we could try to create a temp folder in the current folder as it was before.

Is it make sense?

Reply:
Okay, sounds good.